integration_test/third_party_apps_test.go

```diff
@@ -363,6 +364,75 @@ func runSingleTest(ctx context.Context, logger *logging.DirectoryLogger, vm *gce
 	return nonRetryable, nil
 }
 
+func modifiedFiles(t *testing.T) []string {
+	cmd := exec.Command("git", "diff", "--name-only", "upstream/master")
+	out, err := cmd.Output()
+	if err != nil {
+		t.Fatalf("got error calling `git diff`: %v", err)
+	}
+
+	return strings.Split(string(out), "\n")
+}
+
+func determineImpactedApps(mf []string) map[string]bool {
+	impactedApps := make(map[string]bool)
+	for _, f := range mf {
+		if strings.HasPrefix(f, "apps/") {
+			// File names: apps/<appname>.go
+			f := strings.TrimPrefix(f, "apps/")
+			f = strings.TrimSuffix(f, ".go")
+			impactedApps[f] = true
+
+		} else if strings.HasPrefix(f, "integration_test/third_party_apps_data/applications/") {
+			// Folder names: integration_test/third_party_apps_data/applications/<app_name>
+			f := strings.TrimPrefix(f, "integration_test/third_party_apps_data/applications/")
+			f = strings.Split(f, "/")[0]
+			impactedApps[f] = true
+
+		} else {
+			// Any other modified file means we should test
+			// everything.
+			impactedApps["all"] = true
+			break
+		}
+
+	}
+	return impactedApps
+}
+
+type test struct {
+	platform   string
+	app        string
+	skipReason string
+}
+
+const defaultPlatform = "debian-10"
+
+func determineTestsToSkip(tests []test, impactedApps map[string]bool, testConfig testConfig) {
+	_, testAll := impactedApps["all"]
+	for i, test := range tests {
+		if testing.Short() {
+			if len(impactedApps) > 0 {
+				_, testApp := impactedApps[test.app]
+				if testAll == false && testApp == false {
+					tests[i].skipReason = fmt.Sprintf("skipping %v because it's not impacted by pending change", test.app)
+				} else if testAll == true {
```

Comment:
This differs from what's in the doc. which might be ok. but the difference is, in the doc, certain changes would really trigger all applications on all platforms. For example, changing gce_testing.go would trigger everything. What you have here lacks that option, instead it will test all applications only on the default platform. Not sure if you intended this difference. I'm OK with it though, I can always omit -test.short and run the tests manually in that case. Just wanted to make sure you knew about this difference.

Reply:
This was an oversight.  Let me think about what to do about it.

Reply:
I think this is the right thing to do.  I think we'll cover the full test matrix in postsubmit.